### PR TITLE
Update to v4.21.1

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This merge also removes the bot automerge because it has not been a big success in the last several merges.

### CODE REVIEW

**Comment by xylar** [comment]
@tpdownes, could you take a look at this and make sure it looks right to you?  If so, I will proceed with the next 4.21.x releases as soon as I get `libprotobuf` sorted out.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge/protobuf, this is ready to review.  If you have questions about the strange versioning choices made on https://github.com/protocolbuffers/protobuf as I did yesterday, don't hesitate to ask.

Once I get some feedback or the go-ahead here, I will proceed with the next set of releases as soon as the corresponding `libprotobuf` packages are available.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+{% set libprotobuf_version = "3." + minor_micro %}
 
 package:
   name: protobuf
   version: {{ version }}
 
 source:
-  url: https://github.com/protocolbuffers/protobuf/archive/v{{ version }}/protobuf-v{{ version }}.tar.gz
-  sha256: ca983c9d2c8f8c935513642bcc4b2cbc64e4046e0bb16bf2ff893128577ece8c
+  url: https://github.com/protocolbuffers/protobuf/archive/v{{ minor_micro }}/protobuf-v{{ minor_micro }}.tar.gz
```
Comment (xylar) [inline recipe/meta.yaml:10]: As I mentioned in https://github.com/conda-forge/libprotobuf-feedstock/pull/123, I'm afraid that not having version in the URL might mean the bot doesn't detect future releases automatically. I have made myself a watcher of protobuf releases so I should be able to take care of them manually if that's the case.

File: .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml [hunk 2]
```diff
@@ -2,6 +2,8 @@ channel_sources:
 - conda-forge
 channel_targets:
 - conda-forge main
+libprotobuf:
```
Comment (xhochy) [inline .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml:5]: This doesn't look right.

Comment (xylar) [reply .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml:5]: Hmm, I may need to rerender.

Comment (xylar) [reply .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml:5]: No, that doesn't look like it.

Comment (xylar) [reply .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml:5]: @xhochy, any idea why conda-smithy doesn't notice that I've put version constraints on it?  Any suggestions for what I should change?

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml (hunk 2 above)
Comment (xylar) [reply .ci_support/win_64_python3.9.____cpythonpython_implcpython.yaml]: The answer, near as I can discern from playing around is that conda-smithy behaves differently when `version` is included in the constraint than when it is not.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -21,15 +26,15 @@ requirements:
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}                    # [unix]
     - {{ compiler('cxx') }}                  # [unix]
-    - libprotobuf {{ version }}              # [build_platform != target_platform]
+    - libprotobuf {{ lib_major + "." + version }}  # [build_platform != target_platform]
```
Comment (xylar) [inline recipe/meta.yaml:29]: @xhochy, is this a solution you can live with?  It seems to make conda-smithy happy in the sense that it notices that we're pinning a specific version and doesn't include the version from conda-forge-pinning.

**Comment by tpdownes** [comment]
Outwardly these seems like a reasonable solution, but I'm not well-versed in the details of recipes. I can maintain existing ones when versions bump and dependencies change. That said, there is a direct link to what appears to be a Python specific URL with the right version number (for Python) in it:

https://github.com/protocolbuffers/protobuf/archive/v21.1/protobuf-python-4.21.1.tar.gz

Not sure if that helps us here.

**Review by xhochy — APPROVED** [review]
Fine with me! Thanks!

**Comment by xylar** [comment]
@tpdownes, I saw that as well and experimented with it. The file there isn't appreciably smaller and only has somewhat fewer files so I decided not to change things here without a clear reason for doing it.  My main concern was that the bot might get confused.

Thanks for your thoughts.

**Pull request merged**